src/doc/telio_events.md

```diff
@@ -0,0 +1,150 @@
+
+    - null - If it’s null or missing it means that the no-link detection mechanism is disabled
+    - down - Link state is down reported as down if the state is different from connected or if there the no-link detection mechanism has detected that the link might be down
+    - up - Reported only if the node is connected and there are no clear indications that the link may be down
+
+- is_exit - It is expected that all traffic will be redirected to this node (true if exit node or VPN node)
+
+- is_vpn - Node represents VPN server (true if VPN node only)
+
+- ip_addresses - Address(es) assigned to a node.
+
+- allowed_ips - Indicates subnets being routed to this node.
+
+- hostname - DNS name of a node.
+
+- path - Indicates whether the connection is direct or relay’ed.
+
+- endpoint - The IP address which is used for underlying communication with the node. May be localhost, which indicates relayed connection.
+
+### When node event is reported?
+
+Any change to any property of the node (be it a VPN node or a mesh node) will result in an event being fired.
+
+### Common scenarios:
+
+- VPN flow (node is not in meshnet config, identified by `public_key`):
+
+    - Connection to VPN server is `requestconnect_to_exit_node(...)` called→ `{ "state": "connecting", "is_vpn": true, "is_exit": true, ... }`
+    - Connection to VPN server is established → `{ "state": "connected", ... }`
+    - Connection to VPN server is lost → `{ "state": "connecting", ... }`
+    - Disconnection requested by calling `disconnect_from_exit_node[s](...)` → `{ "state": "disconnected", ... }`
+
+- MESH flow (node is in meshnet config):
+
+    - Enable meshnet by calling `set_meshnet(...)` -> `{ "state": "connecting", "is_vpn": false, "is_exit": false, ...}`
+    - Meshnet node connection established → `{ "state": "connected", ... }`
+    - Meshnet node connection lost → `{ "state": "connecting", ... }`
+    - Node is promoted to be an exit node by calling `connect_to_exit_node(...)` → `{ "is_exit": true, ... }`
+    - Node is demoted to be meshnet node by calling `disconnect_from_exit_node[s](...)` → `{ "is_exit": false, ... }`
+    - Meshnet disabled by calling `set_meshnet_off(...)` → `{ "state": "disconnected", ... }`
+
+## Telio Relay Event
+
+Relay event messages are formed and sent every time relay state changes or start connecting to a different relay server. When a relay event occurs it send the Derp Server json with it current state.
+
+Format of the [Relay][telio_model::config::Server] event:
+
+```json
+{
+  "region_code": string,
+  "name": string,
+  "hostname": string,
+  "ipv4": string,
+  "relay_port": int,
+  "stun_port": int,
+  "stun_plaintext_port": int,
+  "public_key": string,
+  "weight": int,
+  "use_plain_text": bool,
+  "conn_state": "<disconnected|connecting|connected>"
+}
+```
+
+### When node event is reported?
+
+A relay event is triggered on when it changes state or we start connecting to another relay server.
```

Comment:
nit: This sounds to me a little bit strange - something like 'we report an event, when the event changes'. I think it would be better to explicitly state that we report it when we change servers or when the configuration of the server changes.